pydarn/plotting/maps.py

```diff
@@ -1030,3 +1030,117 @@ def plot_potential_contours(cls, fit_coefficient: list, lat_min: list,
                     color=pot_minmax_color, zorder=5.0)
         plt.scatter(np.radians(min_mlon), min_mlat, marker='_', s=70,
                     color=pot_minmax_color, zorder=5.0)
+        
+    
+
+    class TimeSeriesParams(Enum):
+        """
+        Enum class to hold the parameters that can be plotted
+        Members
+        ------------
+        NUM_VECTORS:
+        the number of aviliable vectors in the map
+        IMF_BY:
+        The magnitude of the IMF By component in nT
+        IMF_BZ:
+        The magnitude of the IMF Bz component in nT
+        KP:
+        The Kp index
+        HMB_MAX:
+        The farthest equatorward extent of the HMB
+
+        """
+        NUM_VECTORS = 'num_vectors'
+        IMF_BY = 'imf_by'
+        IMF_BZ = 'imf_bz'
+        KP = 'kp'
+        HMB_MAX = 'hmb_max'     
+        
+    @classmethod
+    def plot_time_series(cls, dmap_data: List[dict],
+                     parameter: Enum = TimeSeriesParams.NUM_VECTORS,
+                     start_record: int = 0,end_record: int = 1, start_time: dt.datetime = None,
+                     end_time: dt.datetime = None):
+        '''
+        Plot time series of various map parameters
+        
+        Params
+        ----------
+        dmap_data: List[dict]
+            List of dictionaries containing the data to be plotted
+        start_record: int
+            time index that we will plot from
+        end_record: int
+            time index that we will plot to (-1 will plot the entire period)
+        start_time: dt.datetime
+            Start time of the data
+        end_time: dt.datetime
+            End time of the data
+            '''
+        #create a new matplotlib figure
+        fig = plt.figure()
+        #if end_record is -1, set it to the last record
+        if end_record == -1:
+            end_record = len(dmap_data)-1
+        #determine the start and end record
```

Comment:
I think maybe for clarity, we should do the same as rtp.py time series plotting does where if the user doesn't put in a start and end time, then we do the whole time frame of data read in (in __determine_start_end_time in rtp.py at the bottom).
Are you using the start/end record here to get around not having the start_time/end_time working, or is that a feature you would like to be able to do? We also implemented an either/or sort of deal that we used in fan.py plot_fan where the user can put in either a record number or a datetime object in for `scan_index: Union[int, dt.datetime] = 1` then on line 181 in fan.py there is `if isinstance(scan_index, dt.datetime):` to determine what to do (obviously here we would have two keywords to consider). I'm not sure if it might just easier to get the start/end time working properly (but it worked for me anyway so most of the way there) as it makes more sense that a user would know the time they want, but probably not what record in the file that corresponds to.